### PR TITLE
OVS-4394 - OVS setup improvements

### PR DESCRIPTION
**Pull request by eduardBM**
- fixed issues with rollback

### CODE REVIEW

File: ovs/lib/setup.py [hunk 1]
```diff
@@ -1056,24 +1027,73 @@ def _rollback_setup(target_client, first_node):
...
+            if first_node is True:
+                for key in EtcdConfiguration.base_config.keys() + ['install_time', 'plugins']:
+                    try:
+                        EtcdConfiguration.delete(key='/ovs/framework/{0}'.format(key))
+                    except (EtcdKeyNotFound, EtcdConnectionFailed, KeyError):
+                        pass
 
-            for key in EtcdConfiguration.base_config.keys() + ['install_time', 'plugins', 'hosts/{0}'.format(machine_id)]:
+            try:
+                EtcdConfiguration.delete(key='/ovs/framework/{0}'.format('hosts/{0}'.format(machine_id)))
```
Comment (khenderick) [inline ovs/lib/setup.py]: Can we change this to `...key='/ovs/framework/hosts/{0}'.format(machine_id)...`?

File: ovs/lib/setup.py [hunk 2]
```diff
@@ -1056,24 +1027,73 @@ def _rollback_setup(target_client, first_node):
...
+            except (EtcdKeyNotFound, EtcdConnectionFailed, KeyError):
+                pass
+
+            #  Arakoon and ETCD must be the last services to be removed
+            for service in ['memcached', 'arakoon-ovsdb']:
```
Comment (khenderick) [inline ovs/lib/setup.py]: It's possible that the `arakoon-ovsdb` service won't exist if an external arakoon cluster was used.

Comment (eduardBM) [inline ovs/lib/setup.py]: It is checked on next line. (if ServiceManager.has_service...)

File: ovs/lib/setup.py (hunk 1 above)
Comment (eduardBM) [inline ovs/lib/setup.py]: Indeed...

File: ovs/lib/setup.py (hunk 2 above)
Comment (khenderick) [inline ovs/lib/setup.py]: Yeah, but I mean that it can have a different name. See a bunch of lines below on how to retrieve the correct name.

**Pull request merged**